### PR TITLE
REF: use BlockManager.apply in csv code

### PR DESCRIPTION
**Pull request by jbrockmendel**
This doesn't get rid of internals usage entirely, but makes that access 1 layer less deep

### CODE REVIEW

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -588,7 +588,7 @@ def astype(self, dtype, copy: bool = False, errors: str = "raise"):
 
                 # use native type formatting for datetime/tz/timedelta
                 if self.is_datelike:
-                    values = self.to_native_types()
+                    values = self.to_native_types().values
```
Comment (jreback) [inline pandas/core/internals/blocks.py:591]: .to_numpy() ?

File: pandas/core/internals/blocks.py [hunk 2]
```diff
@@ -2403,7 +2404,8 @@ def fillna(self, value, **kwargs):
     def to_native_types(self, na_rep="NaT", **kwargs):
```
Comment (jreback) [inline pandas/core/internals/blocks.py:2404]: can you now type all of these to return a Block?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/internals/blocks.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/internals/blocks.py:591]: no, to_native_types returns a Block

File: pandas/core/internals/blocks.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/internals/blocks.py:2404]: sure

Comment (jbrockmendel) [reply pandas/core/internals/blocks.py:2404]: actually adding this makes mypy complain about signature mismatch, rather not futz with it ATM

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks! A small comment about moving the apply call into managers.py

File: pandas/io/formats/csvs.py [hunk 3]
```diff
@@ -319,21 +319,18 @@ def _save_chunk(self, start_i: int, end_i: int) -> None:
...
-                # self.data is a preallocated list
-                self.data[col_loc] = col
+        res = mgr.apply(
+            "to_native_types",
+            na_rep=self.na_rep,
+            float_format=self.float_format,
+            decimal=self.decimal,
+            date_format=self.date_format,
+            quoting=self.quoting,
+        )
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/io/formats/csvs.py:331]: Go for it

[the remark this answers is not in the document]
Comment (jorisvandenbossche) [reply pandas/io/formats/csvs.py:331]: I think that fits perfectly in this PR

**Comment by jreback** [comment]
@jbrockmendel needs a rebase, but I agree @jorisvandenbossche suggsetion might be good in this PR

**Comment by jbrockmendel** [comment]
just rebased.  i encourage @jorisvandenbossche to push the edit he has in mind

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jbrockmendel 

yeah agree that followups to push down to_native_types makes sense

**Pull request merged**